packages/haul-core/src/preset/utils/getSourceMapPlugin.ts

```diff
@@ -0,0 +1,31 @@
+import webpack from 'webpack';
+import {
+  NormalizedServerConfig,
+  NormalizedBundleConfig,
+} from '../../config/types';
+
+export default function getSourceMapPlugin(
+  bundleConfig: NormalizedBundleConfig,
+  serverConfig: NormalizedServerConfig
+) {
+  const baseOptions = {
+    test: /\.(js|css|(js)?bundle)($|\?)/i,
+    filename: '[file].map',
+    moduleFilenameTemplate: '[absolute-resource-path]',
+    module: true,
+  };
+
+  if (bundleConfig.sourceMap === 'inline') {
+    return new webpack.EvalSourceMapDevToolPlugin({
+      ...baseOptions,
+      publicPath: `http://${serverConfig.host}:${serverConfig.port}/`,
+    } as webpack.EvalSourceMapDevToolPluginOptions);
+  } else if (bundleConfig.sourceMap) {
+    return new webpack.SourceMapDevToolPlugin({
+      ...baseOptions,
+      publicPath: `http://${serverConfig.host}:${serverConfig.port}/`,
```

Comment:
publicPath is the same, is this expected? if so, you could fit it into baseOptions